### PR TITLE
🐛 Keen destination: fix for timestamp inference for complex types

### PR DESCRIPTION
**Pull request by maciej-nedza**
## What
Fixed bug, when destination initialization during syncrhonization was failing for some particular sources which use schema composition for their field's types ( "allOf", "anyOf", "oneOf").
Also race condition causing randomly failing test runs should be finally fixed.

## How
Simplified process of timestamp inference, now the map containing streams to cursorfFields mappings doesn't keep track of cursor's type. In process of inference, first the cursor is always tried to be parsed into number, if this fails then it's tried to be parsed into string.

## Recommended reading order
1. `x.java`
2. `y.python`

## Pre-merge Checklist

<summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [x] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [x] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [x] Connector version bumped like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</p>

### CODE REVIEW

**Comment by marcosmarxm** [comment]
Thanks @maciej-nedza can you use the **Update connector** checklist to let us know what steps you are? Do you need any assistance or this is already ready to review?

**Comment by maciej-nedza** [comment]
> Thanks @maciej-nedza can you use the **Update connector** checklist to let us know what steps you are? Do you need any assistance or this is already ready to review?

Hey @marcosmarxm , yes, it was already ready for the review. I've updated the checklist and bumped the version.

File: airbyte-config/init/src/main/resources/seed/destination_definitions.yaml [hunk 1]
```diff
@@ -88,12 +88,12 @@
 - destinationDefinitionId: 8aaf41d0-f6d2-46de-9e79-c9540f828142
   name: Keen
   dockerRepository: airbyte/destination-keen
-  dockerImageTag: 0.1.0
+  dockerImageTag: 0.2.0
```
Comment (maciej-nedza) [inline airbyte-config/init/src/main/resources/seed/destination_definitions.yaml:91]: I've bumped versions here and in the `<uuid>.json` file, however I don't see updated version number in settings -> Destination. Should I do something more ?

Comment (sherifnada) [reply airbyte-config/init/src/main/resources/seed/destination_definitions.yaml:91]: instances don't autoupdate versions for stability reasons. The user must update it by hand. Versions specified in this/the JSON files are used the very first time Airbyte runs.

**Review by sherifnada — APPROVED** [review]
Thanks @maciej-nedza !

**Pull request merged**